experimental/devices/unicornhd/unicornhd.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package unicornhd implements interfacing code to Pimoroni's Unicorn HD hat.
+package unicornhd
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"image/draw"
+
+	"periph.io/x/periph/conn/spi"
+)
+
+const (
+	Height = 16
+	Width  = 16
+	speed  = 9000000
+	bits   = 8
+	prefix = 0x72
+)
+
+// Dev represents a Unicorn HAT HD (https://shop.pimoroni.com/products/unicorn-hat-hd)
+// connected over a SPI port.
+type Dev struct {
+	// Communication
+	connector spi.Conn
+	pixels    *image.NRGBA
+	txBuffer  []byte
+	err       error
+}
+
+// New returns a unicornHD driver that communicates over SPI.
+//
+// The SPI port speed must be 9MHz and the SPI mode, 0, as in the
+// python example library.
+func NewUnicornhd(port spi.Port) (*Dev, error) {
+	connector, err := port.Connect(speed, spi.Mode0, bits)
+	if err != nil {
+		return nil, err
+	}
+	return &Dev{
```

Comment:
define with and height as local variables here, so something like:
```
h := 16
w := 16
return &Dev {
  connector: connector,
  img: image.NewNRGBA(image.Rect(0, 0, w, h)),
  txBuffer: make([]byte, h*w*3+1),
```

The reason I ask for this is that then lines 86 and 89 shouldn't use the constant but a the wide of the image, so that if there's a v2 with 32 pixels wide, the code will be easy to generalize.